Source/Core/DolphinNoGUI/GLFW.h

```diff
@@ -0,0 +1,211 @@
+// Copyright 2016 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#if HAVE_GLFW
+#include <GLFW/glfw3.h>
+#include <unistd.h>
+
+#include "Common/MsgHandler.h"
+
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+#include "Core/State.h"
+
+#include "DolphinNoGUI/Platform.h"
+
+static void OnError(int error, const char* description)
+{
+  fprintf(stderr, "GLFW: 0x%08x: %s\n", error, description);
+}
+
+static void OnWindowClose(GLFWwindow* window)
+{
+  s_shutdown_requested.Set();
+}
+
+static void OnWindowFocus(GLFWwindow* window, int focused)
+{
+  if (focused)
+  {
+    rendererHasFocus = true;
+    if (SConfig::GetInstance().bHideCursor && Core::GetState() != Core::CORE_PAUSE)
+      glfwSetInputMode(window, GLFW_CURSOR, GLFW_CURSOR_HIDDEN);
+  }
+  else
+  {
+    rendererHasFocus = false;
+    if (SConfig::GetInstance().bHideCursor)
+      glfwSetInputMode(window, GLFW_CURSOR, GLFW_CURSOR_NORMAL);
+  }
+}
+
+static void OnKey(GLFWwindow* window, int key, int scancode, int action, int mods)
+{
+  if (action != GLFW_PRESS)
+    return;
+
+  if (key == GLFW_KEY_ESCAPE)
+  {
+    if (Core::GetState() == Core::CORE_RUN)
+    {
+      if (SConfig::GetInstance().bHideCursor)
+        glfwSetInputMode(window, GLFW_CURSOR, GLFW_CURSOR_NORMAL);
+      Core::SetState(Core::CORE_PAUSE);
+    }
+    else
+    {
+      if (SConfig::GetInstance().bHideCursor)
+        glfwSetInputMode(window, GLFW_CURSOR, GLFW_CURSOR_HIDDEN);
+      Core::SetState(Core::CORE_RUN);
+    }
+  }
+  else if ((key == GLFW_KEY_ENTER) && (mods & GLFW_MOD_ALT))
+  {
+    GLFWmonitor* monitor = glfwGetWindowMonitor(window);
+    if (monitor)
+    {
+      int last_window_width = SConfig::GetInstance().iRenderWindowWidth;
+      int last_window_height = SConfig::GetInstance().iRenderWindowHeight;
+      glfwSetWindowMonitor(window, nullptr, GLFW_DONT_CARE, GLFW_DONT_CARE, last_window_width,
+                           last_window_height, GLFW_DONT_CARE);
+    }
+    else
+    {
+      monitor = glfwGetPrimaryMonitor();
+      const GLFWvidmode* vidmode = glfwGetVideoMode(monitor);
+      glfwSetWindowMonitor(window, monitor, 0, 0, vidmode->width, vidmode->height,
+                           vidmode->refreshRate);
+    }
+  }
+  else if (key >= GLFW_KEY_F1 && key <= GLFW_KEY_F8)
+  {
+    int slot_number = key - GLFW_KEY_F1 + 1;
+    if (mods & GLFW_MOD_SHIFT)
+      State::Save(slot_number);
+    else
+      State::Load(slot_number);
+  }
+  else if (key == GLFW_KEY_F9)
+    Core::SaveScreenShot();
```

Comment:
There are blocks with braces, so those should have braces too (even if they're just one-liners)